singer_sdk/helpers/_typing.py

```diff
@@ -222,6 +222,9 @@ def conform_record_data_types(  # noqa: C901
                 boolean_representation = None
             elif elem == 0:
                 boolean_representation = False
+            elif isinstance(elem, str):
+                if elem.lower() in ['false', 'f']:
+                    boolean_representation = False
```

Comment:
@jlloyd-widen what happens if the value is a string but not one of the checked options?

Reply:
then it would force it to true ... so that's not good either

Reply:
@jlloyd-widen even if `True` was an acceptable value, it seems to me that `boolean_representation` wouldn't have a value because there's no `else` here?

Reply:
@edgarrmondragon just added a new commit. I tried to make the logic inclusive of `true` scenarios. Where the input element isn't obviously true or false, I pass it along as the final property value unaltered.